src/main/java/at/ac/tuwien/kr/alpha/grounder/atoms/EnumerationAtom.java

```diff
@@ -67,6 +67,29 @@ public EnumerationAtom substitute(Substitution substitution) {
 		return new EnumerationAtom(super.substitute(substitution).getTerms());
 	}
 
+	/**
+	 * Tests whether the given atom is true under a given substitution. An EnumerationAtom is true under a substitution iff, for the substituted atom the
+	 * following holds true: ENUMERATIONS.get(atom.terms[0]).get(atom.terms[1]).equals(atom.terms[2])
+	 * 
+	 * @param atom
+	 * @param subst
+	 * @return a boolean value indicating if the atom is true under the substitution
+	 */
+	public static boolean isTrueUnderSubstitution(EnumerationAtom atom, Substitution subst) {
+		Term idTerm = atom.getTerms().get(0).substitute(subst);
+		if (!ENUMERATIONS.containsKey(idTerm)) {
+			return false;
+		}
+		Map<Term, Integer> idxMappings = ENUMERATIONS.get(idTerm);
+		Term enumTerm = atom.getTerms().get(1).substitute(subst);
+		if (!idxMappings.containsKey(enumTerm)) {
+			return false;
+		}
+		Term idxTerm = atom.getTerms().get(2).substitute(subst);
+		Integer enumIdx = idxMappings.get(enumTerm);
+		return idxTerm.equals(ConstantTerm.getInstance(enumIdx));
+	}
```

Comment:
This method is a little imprecise:

1. What should happen if the substitution applied to the enumeration atom is not ground? 
2. What happens if a not-yet-seen term is checked but the index of that term fits perfectly with the enumeration, i.e., if the substituted atom is `enum(id1,ter2,13)` and `ENUMERATIONS.get(id1).size() == 12` and `ENUMERATIONS.get(id1).get(ter2) == null`?

Now since this method is only used in the workaround for ensuring that only rules that are satisified are fired during stratified evaluation, I suggest to make it clear in the comment that this is just a helper/workaround for a very specific purpose and not intended for other use.

Reply:
This method has been made obsolete by PR #256 and is now removed.